POC/ColorPicker/ColorPicker/MainWindow.xaml.cs

```diff
@@ -47,21 +54,35 @@ private void OnTransparentScreenClick(object sender, EventArgs e)
             DeactivateColorSelectionMode();
         }
 
-        private void OnColorButtonClick(object sender, EventArgs e)
+        private void OnTransparentScreenEscape(object sender, EventArgs e)
+        {
+            SetColor(_previousColor);
+            DeactivateColorSelectionMode();
+        }
+
+        private void OnKeyDown(object sender, KeyEventArgs e)
         {
-            if (_isColorSelectionEnabled)
+            if (e.Key == Key.Escape)
             {
-                DeactivateColorSelectionMode();
+                OnTransparentScreenEscape(sender, e);
```

Comment:
Consider renaming this function, confusing that main window is calling this for its escape key.

Reply:
In either case, the escape key is used to close the transparent screen, so I think it's still accurate. What do you think?

Reply:
I added an if statement to make this self-documenting.